.github/workflows/reusable-windows-ci.yml

```diff
@@ -0,0 +1,769 @@
+          pushd Fast-DDS-CI
+          # locally master is main
+          git symbolic-ref refs/heads/master refs/heads/main 2>&1 | % $assert
+
+          # selecting specific metas & depends
+          $repo_branch = "${{ inputs.force-meta-branch }}"
+
+          if($repo_branch)
+          {
+            # 1. Externally specified as nightlies
+            git checkout $repo_branch 2>&1 | % $assert
+          }
+          else
+          {
+            # 2. Use heuristics to choose
+            switch -regex ("${{ github.event_name }}")
+            {
+              "pull_request"
+              {
+                  # try: source branch, destination branch
+                  foreach($repo_branch in @("${{ github.head_ref }}", "${{ github.base_ref }}"))
+                  {
+                    if(git checkout $repo_branch 2>&1 | % @checker)
+                    {
+                        $choice = "taking metas & depends from branch $repo_branch"
+                        break;
+                    }
+                    else
+                    {
+                        Write-Output "Fast-DDS-CI lacks branch $repo_branch"
+                    }
+                  }
+              }
+              "push|workflow"
+              {
+                  # use the branch that was push or was selected for the workflow
+                  $repo_branch = "${{ github.ref }}"
+                  if(git checkout $repo_branch 2>&1 | % @checker)
+                  {
+                      $choice = "taking metas & depends from branch $repo_branch"
+                  }
+                  else
+                  {
+                    Write-Output "Fast-DDS-CI lacks branch $repo_branch"
+                  }
+              }
+            }
+
+            # On failure fallback to master
+            if(!$choice)
+            {
+              git checkout master 2>&1 | % $assert
+              $choice = "fallback: taking metas & depends from master"
+            }
+            Write-Output $choice
+          }
+
+          popd
+
+          # Get action credentials for github REST API
+          $secret = ConvertTo-SecureString -String "${{ secrets.GITHUB_TOKEN }}" -AsPlainText
+
+          # Check available queries according with github policy
+          "::group::Rate Limits with github action token"
+          ((Invoke-WebRequest -Authentication OAuth -Token $secret `
+            -Uri https://api.github.com/users/octocat).Headers.GetEnumerator() |
+            ? Key -like "X-RateLimit*") | Out-Host
+          "::endgroup::"
+
+          # download the repos file
+          $repos = ((Invoke-WebRequest -Authentication OAuth -Token $secret `
+            -Uri "${{ github.api_url }}/repos/eProsima/Fast-DDS/contents/fastrtps.repos?ref=${{ github.ref }}" `
+            -Headers @{"method"="GET";"Accept"="application/vnd.github.v3+json"}).content |
+            ConvertFrom-Json).content | ConvertFrom-CIBase64 | ConvertFrom-Yaml
```

Comment:
I'm not sure we actually need this since we are already doing the repository checkout on an earlier step.